src/main/java/org/schemaspy/progress/ConditionalProgress.java

```diff
@@ -0,0 +1,58 @@
+package org.schemaspy.progress;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.time.InstantSource;
+import java.util.concurrent.atomic.LongAdder;
+
+/**
+ * Proc for reporting progressions conditionally
+ * to avoid always reporting progress.
+ * This is to avoid spamming the log.
+ * Unattended don't want to see every progress
+ * and attended just want to know that the
+ * process isn't hanged.
+ * @param <T> exception thrown by the job
+ */
+public class ConditionalProgress<T extends Throwable> implements Proc<T> {
+  private final ProgressReporter progressReporter;
+  private final FinishCallback finishCallback;
+  private final Condition condition;
+  private final Job<T> job;
+  private final InstantSource instantSource;
+
+  public ConditionalProgress(
+      ProgressReporter progressReporter,
+      FinishCallback finishCallback,
+      Condition condition,
+      Job<T> job,
+      InstantSource instantSource
```

Comment:
Five parameters is too much, especially since the Javadoc states that all the object does is reducing the reporting.

Reply:
I think so too but it's a later refactor. I've reached my desired output.